tile-generation/src/main/scala/com/oculusinfo/tilegen/tiling/TileOperations.scala

```diff
@@ -0,0 +1,353 @@
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.oculusinfo.tilegen.tiling
+
+import java.lang.{Double => JavaDouble}
+import java.text.SimpleDateFormat
+
+import com.oculusinfo.binning.impl.AOITilePyramid
+import com.oculusinfo.binning.util.JsonUtilities
+import com.oculusinfo.tilegen.datasets.SchemaTypeUtilities._
+import com.oculusinfo.tilegen.datasets.{CartesianIndexExtractor, CountValueExtractor, _}
+import com.oculusinfo.tilegen.tiling.SparkSqlUtils._
+import com.oculusinfo.tilegen.tiling.analytics._
+import com.oculusinfo.tilegen.util.KeyValueArgumentSource
+
+import scala.collection.JavaConversions._
+
+/**
+ * Provides operations and companion argument parsers that can be bound into a TilePipeline
+ * tree.
+ *
+ */
+// TODO: Provide example of usage in object docs above
+// TODO: Document args for each operation
+// TODO: Add error checking to all operations and parsers
+object TileOperations {
+
+  /**
+   * KeyValueArgumentSource implementation that simply passes the supplied map through.
+   */
+  case class KeyValuePassthrough(args: Map[String, String]) extends KeyValueArgumentSource {
+    def properties = args.map(entry => entry._1.toLowerCase -> entry._2)
+  }
+
+  /**
+   * Registers this standard set of ops
+   */
+  def registerOperations(tilePipeline: TilePipelines) = {
+    // Register basic pipeline operations
+    tilePipeline.registerPipelineOp("json_load", parseLoadCsvDataOp)
+    tilePipeline.registerPipelineOp("csv_load", parseLoadJsonDataOp)
+    tilePipeline.registerPipelineOp("cache", parseCacheDataOp)
+    tilePipeline.registerPipelineOp("date_filter", parseDateFilterOp)
+    tilePipeline.registerPipelineOp("integral_range_filter", parseIntegralRangeFilterOp)
+    tilePipeline.registerPipelineOp("fractional_range_filter", parseFractionalRangeFilterOp)
+    tilePipeline.registerPipelineOp("regex_filter", parseRegexFilterOp)
+  }
+
+  /**
+   * Parse data path from args
+   */
+  def parseLoadJsonDataOp(args: Map[String, String]) = {
+    val argParser = KeyValuePassthrough(args)
+    val path = argParser.getString("ops.path", "HDFS path to data")
+    loadJsonDataOp(path)_
```

Comment:
Why this syntax instead of:
    loadJsonDataOp(path)(_)?
I find the bare _ at the end confusing - I would prefer always to use parentheses except with operators.  But I can be convinced if you have a reason.